File: python/1-bit-and-2-bit-characters.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        i = 0
        is_one_bit = False
        while i < len(bits):
            if bits[i] == 1:
                i += 2
                is_one_bit = False
            else:
                i += 1
                is_one_bit = True
        return is_one_bit

File: rust/1-bit-and-2-bit-characters.rs
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let mut i = 0;
        let mut is_one_bit = false;
        while i < bits.len() {
            if bits[i] == 1 {
                i += 2;
                is_one_bit = false;
            } else {
                i += 1;
                is_one_bit = true;
            }
        }
        return is_one_bit;
    }
}
